core/domain/taskqueue_services.py

```diff
@@ -0,0 +1,117 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Provides a shim for taskqueue-related operations."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+import json
+
+from core.platform import models
+import feconf
+
+platform_taskqueue_services = models.Registry.import_taskqueue_services()
+
+
+# NOTE: The following constants should match the queue names in queue.yaml.
+# Taskqueue for backing up state.
+QUEUE_NAME_BACKUPS = 'backups'
+# Taskqueue for running continuous computation jobs.
+QUEUE_NAME_CONTINUOUS_JOBS = 'continuous-jobs'
+# Default queue for processing tasks (including MapReduce ones).
+QUEUE_NAME_DEFAULT = 'default'
+# Taskqueue for sending email.
+QUEUE_NAME_EMAILS = 'emails'
+# Deferred queue for processing events outside the request/response cycle.
+QUEUE_NAME_EVENTS = 'events'
+# Taskqueue for running one-off jobs.
+QUEUE_NAME_ONE_OFF_JOBS = 'one-off-jobs'
+# Taskqueue for updating stats models.
+QUEUE_NAME_STATS = 'stats'
+
+# Function identifiers inform the deferred task handler of which deferred
+# function should be run for the relevant task.
+# NOTE for developers: If you want to defer a function, visit the file
```

Comment:
... defer a function (i.e., run it asynchronously), visit ...

Reply:
Done.